### PR TITLE
POC - Just for discussion - Move all runner code except DB specific dependencies into FluentMigrator.Runner.Core

### PR DESCRIPTION
**Pull request by gliljas**
This leaves FluentMigrator.Runner as a meta package.

Getting this to work with obsolete code was not worth it in a POC, so I removed it. Ran the integration tests for SQL Server 2016. All green.

### CODE REVIEW

**Comment by jzabroski** [comment]
I'm going to start reviewing this next weekend (4/11-4/12).  Thanks for putting this together, @gliljas !

**Comment by jzabroski** [comment]
Hey @gliljas 

I thought about this idea some more and I really like it.  The economic value it adds is:

1. ability to write Roslyn Analyzers that suggest database-specific best practices based on which Runners the users import.
2. ability to track via nuget.org the popularity of database-specific runners and therefore where to focus efforts of FluentMigrator improvements by popularity (not that popularity is the only consideration, as die hard/passionate users are often the best ways to build community and improve projects - @fubar-coder , @eloekset  and myself fall in the die hard camp and that's how we all got involved - we just couldn't stand to see the project not be actively maintained)
3. ability to split out unit tests per runner - I started doing this as part of #1218 - with the major advantage being that if some vendors/projects fall behind in supporting the .NET releases, the core branch isn't over-complicated with `#if NETFRAMEWORK` and other pre-processor directives and so on.

Previously, nobody could give me great reason to do this other than "I hate files in my directory".

Remaining proof-of-concept is re-bundling FluentMigrator.Console and verifying it ships all runners still.  I am assuming you expect FluentMigrator.DotNet.Cli to also ship everything, and that this change is mainly to support using the In-Process Runner?

**Comment by gliljas** [comment]
Yes, the console app could/should definitely be a more complete package.

**Comment by jzabroski** [comment]
Yes.  Realistically it might take me a few months to factor in this change, as I'm working on restructuring all the tests so that they're split out per runner as well, which is a ton of work.  I'm not a huge fan of injecting all the runners into a single test.  Ideally, there should be a driver program that accepts a runner config as an injected parameter, and the SUT is the path through the runner we wish to exercise.

File: samples/FluentMigrator.Example.Migrator/Program.cs [hunk 1]
```diff
@@ -40,10 +40,6 @@ static int Main(string[] args)
             var app = new CommandLineApplication();
 
             var help = app.HelpOption();
-            var mode = app.Option<MigrationMode>(
```
[the remark this answers is not in the document]
Comment (gliljas) [reply samples/FluentMigrator.Example.Migrator/Program.cs:43]: To be honest, I'm not sure. I believe it was just a victim of making things compile, i.e no legacy stuff.

**Comment by SomebodyOdd** [comment]
@gliljas @jzabroski 
Sorry for bumping, but are there any news on this? Any hope of this getting to a release soon™? =)

**Comment by jzabroski** [comment]
@SomebodyOdd Not a priority right now.  I am trying to true up FluentMigrator develop and master branches and end FluentMigrator v3.x so we can focus on 4.x and onwards.  There is pretty significant drift between the two branches and now that .NET 5 is released, that needs to be the main focus.

I think once we clean up and retire the develop branch, we can more easily integrate a change like this (which would be a massive breaking change).  It would really discourage me personally from ever syncing develop and master if we added a change like this now.  #1348 contains the spreadsheet where I have tried to figure out this drift that started with v3 release.

**Comment by jzabroski** [comment]
To be clear, I do want to do this - along with Async All the Way, and a few other features.  I just have a roadmap in my head of how to get there most efficiently and how to value my free time such that I don't get discouraged along the way.

**Comment by SomebodyOdd** [comment]
@jzabroski Got it, thanks

**Comment by mattbrailsford** [comment]
@jzabroski is there anything you ned to help with to get this over the line? I was contemplating re-submitting it and seeing if we can re-instate much of the deleted code as this appears to be the greatest concern in the code review. Is there anything else that is causing a roadblock?

**Comment by mattbrailsford** [comment]
@jzabroski I thought I'd take a look to see how hard it would be to just move all the code with no deletions and I managed to get it to build and have all tests pass. I've re-submitted it as a new PR #1600

**Pull request closed without merging**